---
title: "Glucosensing cells in the DVC"
author: "Binbin Yan"
date: "March, 2022"
output:
  html_document:
    df_print: paged
    code_folding: hide
  pdf_document: default
---



# Introduction
This file intends to illustrate how the single-nucleus RNA-sequencing (snRNA-seq) transcriptomics data in the dissertation were produced.
The original snRNA-seq datasets used by the analysis were acquired from Ludwig et al.(2021)(GEO accession number:GSE166649) and Dowsett et al.(2021) (GEO accession number: GSE168737).

### References:
1. Dowsett, G.K.C., Lam, B.Y.H., Tadross, J.A., Cimino, I., Rimmington, D., Coll, A.P., Polex-Wolf, J., Knudsen, L.B., Pyke, C. & Yeo, G.S.H. (2021) A survey of the mouse hindbrain in the fed and fasted states using single-nucleus RNA sequencing. Mol Metab, 53, 101240.

2. Ludwig, M.Q., Cheng, W., Gordian, D., Lee, J., Paulsen, S.J., Hansen, S.N., Egerod, K.L., Barkholt, P., Rhodes, C.J., Secher, A., Knudsen, L.B., Pyke, C., Myers, M.G., Jr. & Pers, T.H. (2021) A genetic map of the mouse dorsal vagal complex and its role in obesity. Nat Metab, 3, 530-545.

# Abstract

Dorsal vagal complex (DVC) is comprised of area postrema (AP), nucleus tract solitarius (NTS) and dorsal motor nucleus of vagus (DMNX). Neurons in the DVC can easily access the signals from the blood stream due to the leaky blood barrier, with AP and NTS integrating numerous visceral signals important for body homeostasis and DMNX being an effector exerting regulatory action via vagal efferents. Studies showed neurons in the DVC especially the NTS are responsive to either increase or decrease in extracellular glucose concentration. Utilising snRNA-seq datasets built from the DVC cells by recent studies, this analysis reviewed and reconciled findings from previous studies on DVC glucosensing, and reveal potential subpopulations of glucosensing neurons.

1) Gck and Glu2 separation
2) GABAergic neurons 
3) CA neurons NPY, Glp1R
4) Cholinergic - novel



```{r setting, message=FALSE}
knitr::opts_chunk$set(message = FALSE,warning=FALSE)
```
# Setting up

Load required packages

```{r libraries, message=FALSE,echo=TRUE}
library(Seurat)
library(dplyr)
# conversion from ENSEMBL to gene names requires a AnnotationDbi dataset
library(org.Mm.eg.db)
library(sctransform)
library(ggplot2)
library(ggfittext)
library(tidyverse)
library(kableExtra)
```
Load the Ludwig dataset (AP+NTS+DMNX) with metadata analysed and attached, remove a doublet with count 16 Slc2a2 (Glut2) identified previously, switch the identities of the cells to cell types identified with marker genes.

```{r ludwig dataset load, cache=TRUE,cache.lazy = FALSE}
newDVC <- read_rds("10 DVN_integrated_original metadata and code_clustered.rds")

Idents(newDVC) <- "cell.type"
DefaultAssay(newDVC) <- "RNA"
newDVC <- subset(newDVC,Slc2a2<16,slot="counts")
```


# Overview

A overview of the DVC dataset. The dimplot shows the cell type and number of cell in the labels.
together, there are 72,127 cells, among which are 49,391 neurons and 22,736 glial cells.

```{r dimplot and cluster cell count }
type_num <- str_c(Idents(newDVC),": ",table(Idents(newDVC))[Idents(newDVC)])
newDVC$type_num <- type_num
Idents(newDVC) <- type_num
DimPlot(newDVC,label.size=3.5,label=T,repel=T)+labs(title="Clusters of the DVC dataset")+NoLegend()
Idents(newDVC) <- "cell.type"

# knitr::kable(table(Idents(newDVC)),col.names=c("Cell Clusters","Count"),caption="number of cells per cluster")

```

 
Showing the plots for Glut2 among different cell types.

**The table shows low expression of Glut2 over clusters, and Bar plot shows neurons have the highest percentage of Glut2 cells.**
\
```{r glut2 expression and percentages, cache= TRUE,cache.lazy = FALSE} 

DefaultAssay(newDVC) <- "RNA"
FeaturePlot (newDVC,feature="Slc2a2",cols=c("grey","red"),label=T,label.size=3,repel=T)+labs(title="Slc2a2 - Glut2 expression")

Idents(newDVC) <- "cell.type"

glut2Count<- GetAssayData(newDVC,slot="counts")["Slc2a2",]
newDVC$glut2Count <- glut2Count
VlnPlot(newDVC, feature ="Slc2a2",flip=T, sort=T)+labs(title="Slc2a2/Glut2 expression")
knitr::kable(table(as.character(Idents(newDVC)),c(glut2Count)))
glut2Pos <- ifelse(GetAssayData(newDVC,slot="counts")["Slc2a2",]>=1,"T","F")
newDVC$glut2Pos <- glut2Pos                   

# total percentage
sum(glut2Count>=1)/dim(newDVC[[]])[1]*100

# individual percentage
glut2table = tibble(Glut2=glut2Count,cell.type=as.character(Idents(newDVC)))
glut2Percentage <- glut2table %>% 
  mutate(glut2Pos = ifelse(Glut2>0,1,0)) %>%
  group_by(cell.type) %>%
  summarise(Glut2_Positive_Cells=sum(glut2Pos),Glut2_percentage=round(Glut2_Positive_Cells/n()*100,3),Total_Cell_Count=n()) %>% arrange(desc(Glut2_percentage))


ggplot(glut2Percentage,mapping=aes(x=reorder(cell.type,-Glut2_percentage),y=Glut2_percentage,fill=cell.type))+
  geom_bar(stat="identity")+geom_text(aes(label = Glut2_percentage, vjust = 0),check_overlap=T)+ scale_x_discrete(guide = guide_axis(n.dodge=3))+xlab("Cell Types")+labs(title="Percentage of Glut2+ cells per cell type")+theme(plot.title = element_text(face="bold"))+NoLegend()
# knitr::kable(glut2Percentage)

```

# Neurons

## Glut2+  Neurons

Neurons were isolated from DVC dataset directly using the cell identity labels. 
We then isolate Glut2+ neurons based on gene counts. Glut2+ neurons then underwent subclustering.\
We also acquired scaled gene counts in order to plot the heatmaps.


```{r Glut2 neurons,cache =TRUE, cache.lazy = FALSE}


DefaultAssay(newDVC) <- "integrated"
dvcNeurons <- subset (newDVC,idents = "Neurons")
DimPlot(dvcNeurons)
dvcNeurons <- FindVariableFeatures (dvcNeurons)
dvcNeurons <- ScaleData(dvcNeurons, verbose = FALSE)
dvcNeurons <- RunPCA(dvcNeurons, npcs = 30, verbose = FALSE)
# ElbowPlot(dvcNeurons)
dvcNeurons<-FindNeighbors(object = dvcNeurons, reduction = "pca",dims=1:20)
dvcNeurons<-FindClusters(dvcNeurons,resolution = 0.3)
dvcNeurons <- RunUMAP(dvcNeurons, reduction = "pca", dims = 1:20)

# 
# DefaultAssay(dvcNeurons) <- "RNA"
# dvcNeurons_glut2 <- subset(dvcNeurons, Slc2a2>0, slot="counts")
# DefaultAssay(dvcNeurons_glut2) <- "integrated"
# dvcNeurons_glut2 <- SCTransform(dvcNeurons_glut2 )
# dvcNeurons_glut2 <- RunPCA(dvcNeurons_glut2, verbose = FALSE)
# dvcNeurons_glut2 <- RunUMAP(dvcNeurons_glut2, dims = 1:30,verbose=F)
# dvcNeurons_glut2<-FindNeighbors(object = dvcNeurons_glut2,dims=1:30,assay="SCT")

dvcNeurons_glut2 <- read_rds("DVC_glut2_neuron.rds")
DefaultAssay(dvcNeurons_glut2) <- 'SCT'
# dvcNeurons_glut2<-FindClusters(dvcNeurons_glut2,resolution = 1.3,verbose=F)
DimPlot(dvcNeurons_glut2,label=T,repel=T,size=3.5)+ggmin::theme_powerpoint()+NoLegend()+labs(title="Clusters of Glut2+ neurons, n=249 (out of 49391)")+theme(plot.title = element_text(face="bold"))

DefaultAssay(dvcNeurons_glut2) <- 'RNA'
dvcNeurons_glut2 <- ScaleData(dvcNeurons_glut2,feature=rownames(dvcNeurons_glut2),verbose = FALSE)

```

### Gck+  Neurons 

Similarly, we isolated 1288 Gck+ neurons. Overview figures were plotted.
The neuronal percentage expression of Gck is higher than that of Glut2.


```{r Gck overview, cache.lazy=FALSE,cache=TRUE}

dvcGck <- read_rds("dvcGck.rds")
table(Idents(dvcGck))


gckCount<- GetAssayData(newDVC,slot="counts")["Gck",]
newDVC$gckCount <- gckCount
VlnPlot(newDVC, feature ="Gck",flip=T, sort=T)+labs(title="Gck expression")
knitr::kable(table(as.character(Idents(newDVC)),c(gckCount)))
gckPos <- ifelse(GetAssayData(newDVC,slot="counts")["Gck",]>=1,"T","F")
newDVC$gckPos <- gckPos


# total percentage
sum(gckCount>=1)/dim(newDVC[[]])[1]*100

# individual percentage
gcktable = tibble(Gck=gckCount,cell.type=as.character(Idents(newDVC)))
gckPercentage <- gcktable %>% 
  mutate(gckPos = ifelse(Gck>0,1,0)) %>%
  group_by(cell.type) %>%
  summarise(Gck_Positive_Cells=sum(gckPos),Gck_percentage=round(Gck_Positive_Cells/n()*100,3),Total_Cell_Count=n()) %>% arrange(desc(Gck_percentage))




ggplot(gckPercentage,mapping=aes(x=reorder(cell.type,-Gck_percentage),y=Gck_percentage,fill=cell.type))+
  geom_bar(stat="identity")+geom_text(aes(label = Gck_percentage, vjust = 0),check_overlap=T)+ scale_x_discrete(guide = guide_axis(n.dodge=3))+xlab("Cell Types")+labs(title="Percentage of Gck+ cells per cell type")+theme(plot.title = element_text(face="bold"))+NoLegend()


DimPlot(dvcGck,label=T,repel=T)+ggmin::theme_powerpoint()+NoLegend()+labs(title="Gck+ Neuron Clusters, total n=1288")+theme(plot.title = element_text(face="bold"))

```


### We began the analysis of the gene expression of the Glut2+ and Gck+ neurons. Genes analysed can be categorised into groups:\
\
- **Neuronal subtype marker genes**:\
Inhibitory:"Slc6a5/Glyt2", "Gad1","Gad2", "Slc32a1/Vgat"\
Excitatory:"Slc17a6/Vglut2","Chat"\
Catecholaminergic:"Th","Ddc","Dbh"\
\
- **Genes related to glucosensing**:
"Npy","Gck","Slc2a2/Glut2","Slc2a3/Glut3","Hk1","Kcnk2/Trek-1","Kcnk3/Task-1","Kcnk9/Task-3","Kcnj8/Kir6.1","Kcnj11/Kir6.2","Prkaa2/Ampk","Abcc8/Sur1","Glp1r","Insr"\
\
- **Genes related to astrocyte-neuron communication**:
"Slc16a1/Mct-1","Slc16a7/Mct-2","P2rx3/P2x3","P2rx4/P2x4","P2rx5/P2x5","P2ry11/P2y11"



```{r preparing for the gene analysis 1}
markers.IN <- c("Slc6a5", "Gad1","Gad2", "Slc32a1")
markers.EN <- c("Slc17a6","Chat")
markers.CA <- c("Th","Ddc","Dbh","Npy")
markers.glucose <- c("Gck","Slc2a3","Hk1","Kcnk2","Kcnk3","Kcnk9","Kcnj8","Kcnj11","Prkaa2","Abcc8","Gcg","Glp1r","Insr")
markers.astro <- c("Slc16a1","Slc16a7","P2rx3","P2rx4","P2rx5","P2ry11")

features_full=c(markers.EN, markers.IN,markers.CA,markers.glucose,markers.astro)
```
\
\
Create labels to make it easier to read from the plots
```{r preparing for the gene analysis 2,class.source = 'fold-show'}
gene_labels=c(
 "Slc17a6" = "VGlut2",
    "Slc6a5" = "GlyT2",
    "Slc32a1" = "VGAT",
    "Abcc8"="Sur1",
    "Kcnk3"="Task1",
    "Kcnk9"="Task3",
    "Kcnk2"="Trek1",
    "Prkaa2"="AMPK",
    "Slc16a1"="Mct-1",
    "Slc16a7"="Mct-2",
   "Slc2a2"="Glut2",
    "Slc2a3"="Glut3",
    "Kcnj8"="Kir6.1",
    "Kcnj11"="Kir6.2")
```



```{r glut2 dotplot, include=False}

DotPlot(dvcNeurons_glut2, features=c(markers.EN,markers.IN,markers.CA))+ scale_x_discrete(
  "Marker genes",
labels = gene_labels)+
ylab("Clusters")+coord_flip()+labs(title="Neuronal marker gene expression")+theme(plot.title = element_text(face="bold"))+scale_fill_gradient2( low = rev(c('#d1e5f0','#67a9cf','#2166ac')), mid = "grey", high = rev(c('#b2182b','#ef8a62','#fddbc7')), midpoint = 0, guide = "colourbar", aesthetics = "fill")

```

###Heatmaps for Glut2+ neurons.\

The difference between the two types of heatmaps is that the first one shows the scaled average expression of a particular gene of that cluster, which will show how strong the expression in this cluster is relative to the other, while the second one shows the scaled expression levels of each gene for individual cells.The two types of heatmaps shall be read comparatively. 

```{r Glut2+ heatmaps }

# heatmap 1, average expression 
cluster.averages.glut2 <- AverageExpression(dvcNeurons_glut2, features=features_full,return.seurat = T,assays="RNA")

DoHeatmap(cluster.averages.glut2, features = features_full, size = 3, draw.lines = FALSE)+scale_y_discrete(
  "Marker genes",
labels = c(
    "Slc17a6" = "VGlut2",
    "Slc6a5" = "GlyT2",
    "Slc32a1" = "VGAT",
    "Abcc8"="Sur1",
    "Kcnk3"="Task1",
    "Kcnk9"="Task3",
    "Kcnk2"="Trek-1",
    "Prkaa2"="AMPK",
    "Slc16a1"="Mct-1",
    "Slc16a7"="Mct-2",
    "Slc2a3"="Glut3",
    "Kcnj11"="Kir6.2",
    "Hk1"="Hexokinase1",
    "Th"="Tyrosine Hydroxylase"
    
  ))+ylab("Clusters")+labs(title="Gene expression of Glut2+ Neurons")+theme(plot.title = element_text(face="bold"))+scale_fill_gradient2( low = rev(c('#d1e5f0','#67a9cf','#2166ac')), high = rev(c('#b2182b','#ef8a62','#fddbc7')), midpoint = 0, guide = "colourbar", aesthetics = "fill")


# heatmap 2, single cell expression 
DoHeatmap(dvcNeurons_glut2, features = features_full, size = 3)+scale_y_discrete(
  "Marker genes",
labels = c(
    "Slc17a6" = "VGlut2",
    "Slc6a5" = "GlyT2",
    "Slc32a1" = "VGAT",
    "Abcc8"="Sur1",
    "Kcnk3"="Task1",
    "Kcnk9"="Task3",
    "Kcnk2"="Trek-1",
    "Prkaa2"="AMPK",
    "Slc16a1"="Mct-1",
    "Slc16a7"="Mct-2",
    "Slc2a3"="Glut3",
    "Kcnj8"="Kir6.1",
    "Kcnj11"="Kir6.2",
    "Hk1"="Hexokinase1",
    "Th"="Tyrosine Hydroxylase"
    
  ))+ylab("Clusters")+labs(title="Gene expression of Glut2+ Neurons")+theme(plot.title = element_text(face="bold"))+scale_fill_gradient2( low = rev(c('#d1e5f0','#67a9cf','#2166ac')), high = rev(c('#b2182b','#ef8a62','#fddbc7')), midpoint = 0, guide = "colourbar", aesthetics = "fill")


```



```{r Gck+ heatmaps}
# square heatmap
cluster.averages.gck <- AverageExpression(dvcGck, features=c (features_full[!features_full %in% "Gck"],"Slc2a2"),return.seurat = T,assays="RNA")

DoHeatmap(cluster.averages.gck, features = c (features_full[!features_full %in% "Gck"],"Slc2a2"), size = 3, draw.lines = FALSE)+scale_y_discrete(
  "Marker genes",
labels = c(
    "Slc17a6" = "VGlut2",
    "Slc6a5" = "GlyT2",
    "Slc32a1" = "vGAT",
    "Abcc8"="Sur1",
    "Kcnk3"="Task1",
    "Kcnk9"="Task3",
    "Kcnk2"="Trek-1",
    "Prkaa2"="AMPK",
    "Slc16a1"="Mct-1",    "Slc16a7"="Mct-2",

    "Slc2a3"="Glut3",
    "Kcnj11"="Kir6.2",
    "Hk1"="Hexokinase1",
    "Th"="Tyrosine Hydroxylase",
    "Slc2a2"="Glut2"
    
  ))+ylab("Clusters")+labs(title="Gene expression of Gck+ Neurons")+theme(plot.title = element_text(face="bold"))+scale_fill_gradient2( low = rev(c('#d1e5f0','#67a9cf','#2166ac')), high = rev(c('#b2182b','#ef8a62','#fddbc7')), midpoint = 0, guide = "colourbar", aesthetics = "fill")


# line heatmap
DoHeatmap(dvcGck, features = c (features_full[!features_full %in% "Gck"],"Slc2a2"), size = 3)+scale_y_discrete(
  "Marker genes",
labels = c(
    "Slc17a6" = "VGlut2",
    "Slc6a5" = "GlyT2",
    "Slc32a1" = "vGAT",
    "Abcc8"="Sur1",
    "Kcnk3"="Task1",
    "Kcnk9"="Task3",
    "Kcnk2"="Trek-1",
    "Prkaa2"="AMPK",
    "Slc16a1"="Mct-1",    "Slc16a7"="Mct-2",

    "Slc2a2"="Glut2",
    "Slc2a3"="Glut3",
    "Kcnj11"="Kir6.2",
    "Hk1"="Hexokinase1",
    "Th"="Tyrosine Hydroxylase"
    
    
  ))+ylab("Clusters")+labs(title="Gene expression of Gck+ Neurons")+theme(plot.title = element_text(face="bold"))+scale_fill_gradient2( low = rev(c('#d1e5f0','#67a9cf','#2166ac')), high = rev(c('#b2182b','#ef8a62','#fddbc7')), midpoint = 0, guide = "colourbar", aesthetics = "fill")
```



### Non-overlapping expression of Gck and Glut2 among all the cells

```{r Gck and Hk1 compare}

AverageExpression(dvcNeurons_glut2,feature=c("Gck","Hk1"),assays="RNA") %>% kbl(caption = "Average expression of Gck and Hk1 among Glut2+ neuron clusters") %>%
  kable_classic(full_width = T, html_font = "Times New Roman")

AverageExpression(dvcNeurons_glut2,feature=c("Gck","Hk1"),assays="RNA",group.by="cell.type") %>% kbl(caption = "Average expression of Gck and Hk1 among Glut2+ neurons") %>%
  kable_classic(full_width = T, html_font = "Times New Roman")

DefaultAssay(dvcNeurons) <- "RNA"
dvcNeurons2 <- ScaleData(dvcNeurons,features=c("Gck","Hk1"))

FeatureScatter(dvcNeurons_glut2,feature1="Hk1",feature2="Gck")+xlab("Glut2")
FeatureScatter(dvcNeurons2,feature1="Hk1",feature2="Gck")+xlab("Glut2")

DefaultAssay(dvcNeurons) <- "RNA"
nonglut2neuron <- subset(dvcNeurons,Slc2a2==0, slot="counts")

Idents(nonglut2neuron) <- nonglut2neuron[["integrated_snn_res.0.3"]]
nonglut2neuron <- subset(nonglut2neuron,downsample=250)

AverageExpression(nonglut2neuron,feature=c("Gck","Hk1"),assays="RNA")

FeaturePlot(dvcNeurons_glut2,c("Gck","Hk1"))

gck_glut2 <-unlist(AverageExpression (dvcNeurons_glut2,feature=c("Gck"),assays="RNA"))
gck_nonglut2 <- unlist(AverageExpression(nonglut2neuron,feature=c("Gck"),assays="RNA"))

# boxplot(gck_glut2, gck_nonglut2)

t.test(gck_glut2,gck_nonglut2)

# rm(nonglut2neuron,dvcNeurons,dvcNeurons2)

```



```{r}

```

